test/runtests.jl

```diff
@@ -677,19 +677,87 @@ Revise.track(CthulhuTestSandbox, normpath(@__DIR__, "sandbox.jl"))
 
     @testset "debuginfo: $debuginfo" for debuginfo in instances(Cthulhu.DInfo.DebugInfo)
         @testset "iswarn: $iswarn" for iswarn in tf
-            @testset "verbose: $verbose" for verbose in tf
+            @testset "hide_type_stable: $hide_type_stable" for hide_type_stable in tf
                 @testset "inline_cost: $inline_cost" for inline_cost in tf
                     io = IOBuffer()
                     Cthulhu.cthulhu_typed(io, debuginfo,
                         src, rt, mi;
-                        iswarn, verbose, inline_cost)
+                        iswarn, hide_type_stable, inline_cost)
                     @test !isempty(String(take!(io))) # just check it works
                 end
             end
         end
     end
 end
 
+@testset "hide type-stable statements" begin
+    let # optimize code
+        _, src, infos, mi, rt, slottypes = @eval Module() begin
+            const globalvar = Ref(42)
+            $process() do
+                a = sin(globalvar[])
+                b = sin(undefvar)
+                return (a, b)
+            end
+        end
+        function prints(; kwargs...)
+            io = IOBuffer()
+            Cthulhu.cthulhu_typed(io, :none, src, rt, mi; kwargs...)
+            return String(take!(io))
+        end
+
+        let # by default, should print every statement
+            s = prints()
+            @test occursin("globalvar", s)
+            @test occursin("undefvar", s)
+        end
+        let # should omit type stable statements
+            s = prints(; hide_type_stable=true)
+            @test !occursin("globalvar", s)
+            @test occursin("undefvar", s)
+        end
+    end
+
+    let # unoptimize code
+        _, src, infos, mi, rt, slottypes = @eval Module() begin
+            const globalvar = Ref(42)
+            $process(; optimize=false) do
+                a = sin(globalvar[])
+                b = sin(undefvar)
+                return (a, b)
+            end
+        end
+        function prints(; kwargs...)
+            io = IOBuffer()
+            Cthulhu.cthulhu_typed(io, :none, src, rt, mi; kwargs...)
+            return String(take!(io))
+        end
+
+        let # by default, should print every statement
+            s = prints()
+            @test occursin("globalvar", s)
+            @test occursin("undefvar", s)
+        end
+        let # should omit type stable statements
+            s = prints(; hide_type_stable=true)
+            @test !occursin("globalvar", s)
+            @test_broken occursin("undefvar", s)
```

Comment:
I found the current implementation of `is_type_unstable` may not work well for unoptimized code.

Especially, it doesn't handle `lhs = rhs` statements where `rhs` is type unstable ?
```julia
julia> const globalvar = Ref(42)
Base.RefValue{Int64}(42)

julia> descend(; optimize=false) do
           a = sin(globalvar[])
           b = sin(undefvar)
           return (a, b)
       end
│ ─ %-1  = invoke #3()::Tuple{Float64, Any}
    @ REPL[4]:2 within `#3`
1 ─ %1 = Base.getindex(Main.globalvar)::Int64
│        (a = Main.sin(%1))::Float64
│   @ REPL[4]:3 within `#3`
│        (b = Main.sin(Main.undefvar))::Any
│   @ REPL[4]:4 within `#3`
│   %4 = Core.tuple(a, b)::Tuple{Float64, Any}
└──      return %4
Select a call to descend into or ↩ to ascend. [q]uit. [b]ookmark.
Toggles: [o]ptimize, [w]arn, [h]ide type-stable statements, [d]ebuginfo, [i]nlining costs, [s]yntax highlight for Source/LLVM/Native.
Show: [S]ource code, [A]ST, [L]LVM IR, [N]ative code
Actions: [E]dit source code, [R]evise and redisplay
Advanced: dump [P]arams cache.
 • %1  = getindex(::Base.RefValue{Int64})::Int64
   %2  = sin(::Int64)::Float64
   ↩

│ ─ %-1  = invoke #3()::Tuple{Float64, Any}
│   @ REPL[4]:4 within `#3`
│   %4 = Core.tuple(a, b)::Tuple{Float64, Any}
Select a call to descend into or ↩ to ascend. [q]uit. [b]ookmark.
Toggles: [o]ptimize, [w]arn, [h]ide type-stable statements, [d]ebuginfo, [i]nlining costs, [s]yntax highlight for Source/LLVM/Native.
Show: [S]ource code, [A]ST, [L]LVM IR, [N]ative code
Actions: [E]dit source code, [R]evise and redisplay
Advanced: dump [P]arams cache.
 • %1  = getindex(::Base.RefValue{Int64})::Int64
   %2  = sin(::Int64)::Float64
   ↩
```

Reply:
Hmm, yeah, we might need to special-case assignments to slots.

Reply:
I found it `idx in used` returns `false` for unoptimized code:
https://github.com/JuliaDebug/Cthulhu.jl/blob/93743cb4efe0ddfd189808ec589449ef97361387/src/codeview.jl#L76

So maybe we need to add support for unoptimized code case within Julia Base ?

Reply:
Hmm, I thought that should already work 🤔. I can look into it later today. Probably doesn't need to block this PR though.